File: DSA/Data Structure/Stack.java
import java.nio.channels.IllegalSelectorException;
import java.util.ArrayList;

public class Stack<T> {

    private ArrayList<T> items;

    public Stack() {
        items = new ArrayList<>();
    }

    public void push(T item) {
        items.add(item);
    }

    public T pop() {
        if (isEmpty()) {
            throw new IllegalStateException("Stack is empty");
        }
        return items.remove(items.size() - 1);
    }

    public T peek() {
        if (isEmpty()) {
            throw new IllegalStateException("Stack is empty");
        }
        return items.get(items.size() - 1);
    }

    public boolean isEmpty() {
        return items.isEmpty();
    }

    public int size() {
        return items.size();
    }

    public boolean search(T item) {
        return items.contains(item);
    }

    public static void main(String[] args) {
        Stack<Integer> stack = new Stack<>();

        stack.push(10);
        stack.push(20);
        stack.push(30);
        stack.push(40);
        
        System.out.println(stack.size());
        System.out.println(stack.search(10));
        System.out.println(stack.search(50));
        System.out.println(stack.peek());
        System.out.println(stack.pop());
        System.out.println(stack.pop());
        System.out.println(stack.size());
    }
}

File: DSA/Data Structure/Stack.py
class Stack:
    def __init__(self):
        self.items = []
    def push(self, item):
        self.items.append(item)
    def pop(self):
        if self.is_empty():
            raise IndexError("Stack is empty")
        return self.items.pop()
    def peek(self):
        if self.is_empty():
            raise IndexError("Stack is empty")
        return self.items[-1]
    def is_empty(self):
        return not self.items
    def size(self):
        return len(self.items)
    def search(self, item):
        return item in self.items


stack = Stack()

stack.push(10)
stack.push(20)
stack.push(30)

print(stack.size())
print(stack.search(10))
print(stack.search(50))
print(stack.peek())   
print(stack.pop())   
print(stack.pop())    
print(stack.size())